pkg/downstream/suggest.go

```diff
@@ -121,3 +122,58 @@ func iteratePullRequestCommits(ctx context.Context, client *github.Client, org,
 			return client.PullRequests.ListCommits(ctx, org, repo, prNum, o)
 		})
 }
+
+func hasCommit(git utils.GitHelper, req *SuggestRequest, found []string, c *github.Commit) (bool, error) {
+	has_commit := false
+	for _, commit := range found {
+		if len(commit) == 0 {
+			return false, nil
+		}
+
+		has, err := compareDiff(git, req, commit, c.GetURL())
+		if err != nil {
+			return false, err
+		}
+		has_commit = has
+	}
+
+	return has_commit, nil
+}
+
+func compareDiff(git utils.GitHelper, req *SuggestRequest, c string, u string) (bool, error) {
+	hash := strings.Split(u, "/commits/")[1]
+	url := "https://github.com/" + req.UpstreamOrg + "/" + req.UpstreamRepo + "/commit/" + hash + ".diff"
```

Comment:
Is there no way of using the github Go client for this?

Reply:
No AFAIK
Even `github.Commit.getSHA()` returns an empty string 🤷
Here I parse the the GitHub API commit URL to get that